samples/dnn/text_detection.cpp

```diff
@@ -159,3 +220,71 @@ void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
         }
     }
 }
+
+void four_points_transform(const Mat& frame, const RotatedRect& box, const Size2f& box_size, const Point2f& ratio,
+                           Mat& result) {
+
+    Point2f vertices[4];
+    box.points(vertices);
+    for (int j = 0; j < 4; ++j)
+    {
+        vertices[j].x *= ratio.x;
+        vertices[j].y *= ratio.y;
+    }
```

Comment:
I thing that we should scale resulting bounding boxes once at `decode_bounding_boxes`